common/app/views/support/cleaner/AmpEmbedCleaner.scala

```diff
@@ -39,14 +39,16 @@ case class AmpEmbedCleaner(article: Article) extends HtmlCleaner {
     })
   }
 
+  class AmpExternalVideo(val videoId: String, val elementType: String) {}
+
   object AmpExternalVideo {
-    def getElementTypeByUrl(videoUrl: String) : String = {
-      val youtubePattern = ".*youtube.com.*".r
-      val vimeoPattern = ".*vimeo.com.*".r
+    def getAmpExternalVideoByUrl(videoUrl: String) : Option[AmpExternalVideo] = {
+      val youtubePattern = ".*https?://www.youtube.com/watch\\?v=(.+).*".r
+      val vimeoPattern = ".*https?://vimeo.com/(\\d+).*".r
       videoUrl match {
-        case youtubePattern() => "amp-youtube"
-        case vimeoPattern() => "amp-vimeo"
-        case _ => ""
+        case youtubePattern(videoId) => Some(new AmpExternalVideo(videoId, "amp-youtube"))
+        case vimeoPattern(videoId) => Some(new AmpExternalVideo(videoId, "amp-vimeo"))
+        case _ => None
```

Comment:
With the case class you can avoid the `new` here.

You could go further and introduce `case class YoutubeExternalVideo(videoId: String) extends AmpExternalVideo("amp-youtube")` or something which would require some other changes... but I think your simple solution is better in this case.